src/snowflake/snowpark/mock/_functions.py

```diff
@@ -929,3 +930,99 @@ def mock_to_variant(expr: ColumnEmulator):
     res = expr.copy()
     res.sf_type = ColumnType(VariantType(), expr.sf_type.nullable)
     return res
+
+
+DATETIME_PART_TO_ALIASES = {
+    "year": {"y", "yy", "yyy", "yyyy", "yr", "years", "yrs"},
+    "quarter": {"q", "qtr", "qtrs", "quarters"},
+    "month": {"mm", "mon", "mons", "months"},
+    "week": {"w", "wk", "weekofyear", "woy", "wy"},
+    "day": {"d", "dd", "days", "dayofmonth"},
+    "hour": {"h", "hh", "hr", "hours", "hrs"},
+    "minute": {"m", "mi", "min", "minutes", "mins"},
+    "second": {"s", "sec", "seconds", "secs"},
+    "millisecond": {"ms", "msec", "milliseconds"},
+    "microsecond": {"us", "usec", "microseconds"},
+    "nanosecond": {
+        "ns",
+        "nsec",
+        "nanosec",
+        "nsecond",
+        "nanoseconds",
+        "nanosecs",
+        "nseconds",
+    },
+}
+
+DATETIME_PARTS = set(DATETIME_PART_TO_ALIASES.keys())
+ALIASES_TO_DATETIME_PART = {
+    v: k for k, l in DATETIME_PART_TO_ALIASES.items() for v in l
+}
+DATETIME_ALIASES = set(ALIASES_TO_DATETIME_PART.keys())
+
+
+def unalias_datetime_part(part):
+    if part in DATETIME_PARTS:
+        return part
+    elif part in DATETIME_ALIASES:
+        return ALIASES_TO_DATETIME_PART[part]
+    else:
+        raise ValueError(f"{part} is not a recognized date or time part.")
```

Comment:
nit:

I think you can add the self mapping entry like {"year":"year"} into ALIASES_TO_DATETIME_PART, so that we just need to check ALIASES_TO_DATETIME_PART all the time